### PR TITLE
passed tests

### PR DESCRIPTION
**Pull request by ssaviak**
[DEMO LINK](https://ssaviak.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by VasiliyYarosh — APPROVED** [review]
Brilliant, good job!

File: src/components/TodoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    idsToUpdate(todoId);
+
+    deleteTodo(todoId)
+      .then(() => setTodos(prevTodos => prevTodos
+        .filter(todoToFilter => todoToFilter.id !== todoId)))
+      .catch(() => setErrorMessage(ErrorMessage.FailedDeleteTodo))
+      .finally(() => {
+        idsToUpdate(null);
+      });
+  };
```
Comment (olya-shyian) [inline src/components/TodoItem.tsx:35]: it looks like you have the same code 3 times
image
image

could you do some code refactoring?

you can keep this in TodoContext and reuse everywhere you need

File: src/components/TodoItem.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    const { completed, id } = todoToStatusUpdate;
+
+    idsToUpdate(id);
+
+    updateTodos({ ...todoToStatusUpdate, completed: !completed })
+      .then(() => {
+        updateTodoList({ ...todoToStatusUpdate, completed: !completed });
+      })
+      .catch(() => setErrorMessage(ErrorMessage.FailedUpdateTodo))
+      .finally(() => idsToUpdate(null));
```
Comment (olya-shyian) [inline src/components/TodoItem.tsx:47]: the same with update
image
image

File: src/contexts/TodoContext.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    errorMessage,
+    filteredTodos: todos.filter(todo => {
+      switch (filters.status) {
+        case 'uncompleted':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return true;
+      }
```
Comment (olya-shyian) [inline src/contexts/TodoContext.tsx:73]: move this function to `utils` folder please

File: src/types/ErrorMessage.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export enum ErrorMessage {
+  FailedLoad = 'Unable to load todos',
+  FailedAddTodo = 'Unable to add a todo',
+  FailedDeleteTodo = 'Unable to delete a todo',
+  FailedUpdateTodo = 'Unable to update a todo',
+  FailedUpdateRequest = 'Unable to update a todo',
```
Comment (olya-shyian) [inline src/types/ErrorMessage.ts:6]: you can delete one
image

**Review by olya-shyian — CHANGES_REQUESTED** [review]
Thank you for the task!🌸

Try to remove this pop up when click on radio button
image

File: src/contexts/TodoContext.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      .catch(() => setErrorMessage(ErrorMessage.FailedDeleteTodo))
+      .finally(() => idsToUpdate(null));
+  }, []);
+
+  const value: TodoContextType = useMemo(() => ({
+    todos,
+    idsToChange,
+    tempTodo,
+    errorMessage,
+    filteredTodos: todos.filter(todo => {
```
Comment (DanilWeda) [inline src/contexts/TodoContext.tsx:80]: need to move this function to another file

**Review by DanilWeda — COMMENTED** [review]
Great! 
Approved.
Thanks.

**Review by DanilWeda — APPROVED** [review]
Approved*

**Pull request closed without merging**